File: TT-02/one.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int isPalindromo(char* vect, int pInicial, int boo, int tamanho) {                        //Retorna um valor inteiro informando se é palíndromo ou não -\- 1 - Palíndromo  |  0 - !(Palíndromo)
        /*
        *
        * A variavel pInicial deve ser instaciada como 0 e a variável boo deve ser instaciada como 1
        *
        */
        if (pInicial >= tamanho / 2)
        return boo;
        else {
        if (vect[pInicial] == vect[(tamanho - 1) - pInicial])
        return isPalindromo(vect, ++pInicial, boo, tamanho);
        else
        return isPalindromo(vect, tamanho - 1, 0, tamanho);
        }
    }

int testeTermino(char* frase){
    if(strlen(frase) == 4){
        if(frase[0] == 'F' && frase[1] == 'I' && frase[2] == 'M'){
            return 0;
        }
        else
            return 1;
    }
    else
        return 1;
}
int main(){
    char frase[100];
    int flag, tamanho;
    do{
        fgets(frase, sizeof(frase), stdin);
        tamanho = strlen(frase);
        setbuf(stdin, NULL);
        if (testeTermino(frase) == 1){
            flag = isPalindromo(frase, 0, 1, strlen(frase)-1);

            if(flag == 0)
                printf("NAO\n");
            else
                printf("SIM\n");
        }
    
    }while(testeTermino(frase) == 1);


return 0;
}

File: TT-02/one.java
class one{
    public static boolean isPalindromo(char[] vect, int pInicial, boolean boo) {                        //Retorna um valor boolean informando se é palíndromo ou não  
        /*
        *
        * A variavel pInicial deve ser instaciada como 0 e a variável boo deve ser instaciada como true
        *
        */
        if (pInicial >= vect.length / 2)
        return boo;
        else {
        if (vect[pInicial] == vect[(vect.length - 1) - pInicial])
        return isPalindromo(vect, ++pInicial, boo);
        else
        return isPalindromo(vect, vect.length - 1, false);
        }
    }
    public static void charFy(char[] stringVetor, String word) {                                        // Transforma a string em um vetor de caracteres
            for (int i = 0; i < word.length(); i++)
                stringVetor[i] = word.charAt(i);
    }

    public static void main(String [] args){
        String frase;
        char[] fraseArray;
        boolean boo;
        do{
            frase = MyIO.readLine();
            if(!(frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M')){
                fraseArray = new char[frase.length()];
                charFy(fraseArray, frase);
                
                boo = isPalindromo(fraseArray, 0, true);
                
                if(boo)
                MyIO.println("SIM");
                else
                MyIO.println("NAO");
            }
        }while(!(frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M'));
    }

}